File: 品味Java的21種設計模式/dp_cpp/strategy/example8/DbLog.cpp
#include "DbLog.h"

namespace cn
{
	namespace javass
	{
		namespace dp
		{
			namespace strategy
			{
				namespace example8
				{

					void DbLog::doLog(std::string msg)
					{
						//制造错误
//JAVA TO C++ CONVERTER TODO TASK: There is no direct native C++ equivalent to the Java String 'trim' method:
						if(msg!="" && msg.trim()->length()>5)
						{
							int a = 5/0;
						}
						puts("现在把 '"+msg+"' 记录到数据库中");
					}
				}
			}
		}
	}
}

File: 品味Java的21種設計模式/dp_java/strategy/example8/DbLog.java
package cn.javass.dp.strategy.example8;
/**
 * 把日志记录到数据库
 */
public class DbLog extends LogStrategyTemplate {
    public void doLog(String msg) {
        //制造错误
        if (msg != null && msg.trim().length() > 5) {
            int a = 5 / 0;
        }

        System.out.println("现在把 '" + msg + "' 记录到数据库中");
    }
}
